src/common/utils/use-waiting-message.ts

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+
+// Keys are the seconds to wait before showing the message
+export type WaitingMessages = {
+  [key: string]: string;
+};
+
+function messageForSecondsPassed(waitingMessages: WaitingMessages, seconds: number) {
+  return waitingMessages[seconds.toString() as keyof typeof waitingMessages];
+}
+
+export function useWaitingMessage(
+  waitingMessages: WaitingMessages,
+  { waitingMessageInterval, maxCheckWaitingTime } = {
+    waitingMessageInterval: 1000,
+    maxCheckWaitingTime: 70000,
+  }
+): [string, () => void, () => void] {
+  let stopped = false;
```

Comment:
The react way of doing this be to use a state hook, too, right? If you're trying to avoid a rerender, you could use `useRef(false)` and update that instead.